include/sys/cred.h

```diff
@@ -0,0 +1,30 @@
+#ifndef _SYS_UCRED_H_
+#define _SYS_UCRED_H_
+
+#include <sys/types.h>
+#include <sys/proc.h>
+#include <stdint.h>
+
+#define NGROUPS_MAX 16
+
+/*
+ * Kernel view of credencials
+ */
+struct cred {
```

Comment:
It would be better to use `typedef` in declaration of struct, e.g.:
```c
typedef struct cred {
...
} cred_t;
``` 
And then use type `cred_t` instead of `struct cred`.